sql/core/src/main/scala/org/apache/spark/sql/execution/python/ArrowEvalPythonExec.scala

```diff
@@ -44,14 +66,24 @@ case class ArrowEvalPythonExec(udfs: Seq[PythonUDF], output: Seq[Attribute], chi
     val schemaOut = StructType.fromAttributes(output.drop(child.output.length).zipWithIndex
       .map { case (attr, i) => attr.withName(s"_$i") })
 
+    val batchSize = conf.arrowMaxRecordsPerBatch
+
+    val batchIter = if (batchSize > 0) {
+      new BatchIterator(iter, batchSize)
+    } else if (batchSize == 0) {
+      Iterator(iter)
+    } else {
+      throw new IllegalArgumentException(s"MaxRecordsPerBatch must be >= 0, but is $batchSize")
```

Comment:
We should handle negative value of `conf.arrowMaxRecordsPerBatch` as the same as `0` according to the doc of the conf:

> When using Apache Arrow, limit the maximum number of records that can be written to a single ArrowRecordBatch in memory. If set to zero or negative there is no limit.

Reply:
Fixed.